### DISCUSSION TITLE
Suggestions on how to create steps in conversation flow

### DISCUSSION THREAD

**Discussion by napon** [trypromptly/LLMStack#23] [Q&A] [answered]
Hi there, 

I'm new to Promptly and have been playing around with creating a chatbot. Really like the platform and what you guys are building! I have a few questions around how to customize the bot further:
1. Are there any documentations on how to configure the OpenAI Processor particularly for the bot to call functions when needed?
2. Are there examples of chatbot implementation that includes conversational steps / prompting user for inputs?

Thank you in advance!

**Comment by ajhai** [comment, marked as answer]
> Are there any documentations on how to configure the OpenAI Processor particularly for the bot to call functions when needed?

We do not yet allow function calling within the app. Right now it simply returns OpenAI's response i.e., function to call along with arguments.

> Are there examples of chatbot implementation that includes conversational steps / prompting user for inputs?

This can be done with some prompting + promptly/text-chat or openai/chatgpt processors. Make sure to turn on history in advanced settings of the processors so the model can remember the conversation flow and ask follow up questions. I can share a template to get started with if that helps.

**Reply by napon** [reply]
Yes if you could share a template on how to accomplish this that would be very helpful! Thank you!
